### PR TITLE
Fixing HttpResponse headers value

### PR DESCRIPTION
**Pull request by ygy**

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/util/executor/HttpExecutorImplTest.java [hunk 1]
```diff
@@ -71,6 +75,36 @@ public void afterMethod() throws Exception {
...
+                .method("GET")
+                .uri(baseUrl.toURI())
+                .build();
+        HttpResponse executorResponse = executor.execute(executorRequest);
+        assertTrue(executorResponse.headers().containsKey(HTTP_RESPONSE_CUSTOM_HEADER_KEY));
+        assertTrue(Iterables.getOnlyElement(executorResponse.headers().get(HTTP_RESPONSE_HEADER_KEY)).equals(HTTP_RESPONSE_HEADER_VALUE));
+
+        assertTrue(executorResponse.headers().containsKey(HTTP_RESPONSE_HEADER_KEY));
+        assertTrue(Iterables.getOnlyElement(executorResponse.headers().get(HTTP_RESPONSE_CUSTOM_HEADER_KEY)).equals(HTTP_RESPONSE_CUSTOM_HEADER_VALUE));
+    }
```
Comment (neykov) [inline core/src/test/java/org/apache/brooklyn/util/executor/HttpExecutorImplTest.java]: The essence of the test is line `90`. You can cut it down to verifying that `httpResponse.headers()` contains the expected values. Everything else just gets that value through a number of hoops before you check it at the end.

File: utils/common/src/main/java/org/apache/brooklyn/util/http/executor/HttpResponse.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public Builder reasonPhrase(@Nullable String val) {
         }
 
         public Builder headers(Multimap<String, String> val) {
-            headers.putAll(headers);
+            headers.putAll(val);
```
Comment (neykov) [inline utils/common/src/main/java/org/apache/brooklyn/util/http/executor/HttpResponse.java]: Nice find :)

**Comment by neykov** [comment]
Could simplify the test, otherwise good to go.

**Comment by ygy** [comment]
@aledsage @neykov The test has been simplified

**Pull request merged**